src/App.tsx

```diff
@@ -1,23 +1,32 @@
-import React, { useState } from 'react';
-// import AddInvoiceElement from './Components/AddInvoiceElement';
+import { getAuth, onAuthStateChanged } from 'firebase/auth';
+import { useEffect, useState } from 'react';
 import HomePage from './Components/HomePage';
-import LoginPage from './Components/LoginPage';
-import ModalWindow from './Components/ModalWindow';
-import SignUpPage from './Components/SignUpPage';
-import { cteateUserInFirebaseWithEmailAndPassword, loginToFirebaseWithEmailAndPassword, logoutFromFirebase } from './dataBaseUtils/auth/authUtils';
+import LoginPage from './Components/loginAndSignup/LoginPage';
+import SignUpPage from './Components/loginAndSignup/SignUpPage';
 import "./dataBaseUtils/auth/firebaseInit.ts"
-import { writeIntoDataBase } from './dataBaseUtils/readWrite';
 
 function App() {
-	const [signedIn, setSignedIn] = useState<boolean>(false);
-	const [currentPage, setCurrentPage] = useState<string>("login");
-	// loginToFirebaseWithEmailAndPassword("treex.lights@gmail.com", "123456");
-	return (
-		<div >
-			{currentPage === "login" && (!signedIn ? <LoginPage setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} /> : <HomePage setSignedIn={setSignedIn} />)}
-			{currentPage === "signup" && (!signedIn ? <SignUpPage setSignedIn={setSignedIn} setCurrentPage={setCurrentPage} /> : <HomePage setSignedIn={setSignedIn} />)}
-		</div>
-	);
+  const [signedIn, setSignedIn] = useState<boolean>(false);
+  const [currentPage, setCurrentPage] = useState<string>("login");
```

Comment:
Same as above.

Reply:
And, by the way, you might want to type this a bit more thorough (depending on future plans). You could have a type:
```
type RoutePath = "login" |  "signup"
```
or
```
enum Route {
  Login: "login",
  SignUp: "signup",
}
```
and use it instead of "just string" in order to get a better type safety.

Reply:
Enum turned out to be the best option. Changed the type from string to enum